craft_application/application.py

```diff
@@ -261,6 +273,40 @@ def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Pat
 
         return (project_dir / f"{self.app.name}.yaml").resolve(strict=True)
 
+    def get_build_plan(
```

Comment:
So are applications expected to override this? For instance rockcraft, that needs to call `get_rock_build_plan()` instead of the default `get_platforms_build_plan()`?

Reply:
Yes - instead of implementing a BuildPlanner class.